src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmIdentification.h

```diff
@@ -74,79 +81,63 @@ namespace OpenMS
-      std::vector<TransformationDescription> &);
-
-    // Docu in base class
-    virtual void setReference(Size reference_index = 0,
-                              const String & reference_file = "");
+    // Set a reference for the alignment
+    template <typename DataType> void setReference(DataType& data)
+    {
+      reference_.clear();
+      if (data.empty()) return; // empty input resets the reference
+      SeqToList rt_data;
+      bool sorted = getRetentionTimes_(data, rt_data);
+      computeMedians_(rt_data, reference_, sorted);
+      if (reference_.empty())
+      {
+        throw Exception::MissingInformation(__FILE__, __LINE__, __PRETTY_FUNCTION__, "Could not extract retention time information from the reference file");
+      }
+    }
 
     /**
-      @brief Align feature maps or consensus maps.
+      @brief Align feature maps, consensus maps, peak maps, or peptide identifications.
 
-      Since the method of aligning feature and consensus maps is equal for this algorithm,
-      alignFeatureMaps and alignConsensusMaps are only defined in conformance with the interface and
-      forward to this method.
-
-      @param maps Vector maps (FeatureMap or ConsensusMap) that should be aligned.
-      @param transformations Vector of TransformationDescription that will be computed.
+      @param data Vector of input data (FeatureMap, ConsensusMap, MSExperiment<> or @p vector<PeptideIdentification>) that should be aligned.
+      @param transformations Vector of RT transformations that will be computed.
     */
-    template <typename MapType>
-    void alignMaps(std::vector<MapType> & maps,
-                   std::vector<TransformationDescription> & transformations)
+    template <typename DataType>
+    void align(std::vector<DataType>& data, 
+               std::vector<TransformationDescription>& transformations,
+               Int reference_index = -1)
     {
-      checkParameters_(maps.size());
+      checkParameters_(data.size());
       startProgress(0, 3, "aligning maps");
 
-      if (reference_index_)     // reference is one of the input files
+      reference_index_ = reference_index;
+      if (reference_index >= 0) // reference is one of the input files
       {
-        SeqToList rt_data;
-        getRetentionTimes_(maps[reference_index_ - 1], rt_data);
-        computeMedians_(rt_data, reference_, true);
+        if (reference_index >= data.size())
+        {
+          throw Exception::IndexOverflow(__FILE__, __LINE__,
+                                         __PRETTY_FUNCTION__, reference_index, 
+                                         data.size());
+        }
+        setReference(data[reference_index]);
       }
 
       // one set of RT data for each input map, except reference:
-      std::vector<SeqToList> rt_data(maps.size() - bool(reference_index_));
-      for (Size i = 0, j = 0; i < maps.size(); ++i)
+      std::vector<SeqToList> rt_data(data.size() - (reference_index >= 0));
```

Comment:
Well, there's a helpful comment above that line...
I can add a `bool use_reference = (reference_index >= 0);`, but I would still subtract the bool to get the correct size. The alternative would be to add an "if" clause and call `rt_data.resize` with two different values, but that seems unnecessarily long-winded.